platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/banner/BannerScope.kt

```diff
@@ -0,0 +1,183 @@
+package org.jetbrains.jewel.ui.component.banner
+
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.remember
+import org.jetbrains.jewel.foundation.GenerateDataFunctions
+import org.jetbrains.jewel.ui.component.DropdownLink
+import org.jetbrains.jewel.ui.component.IconActionButton
+import org.jetbrains.jewel.ui.component.Link
+import org.jetbrains.jewel.ui.component.Text
+import org.jetbrains.jewel.ui.component.items
+import org.jetbrains.jewel.ui.icon.IconKey
+
+/**
+ * Defines the scope to configure actions for a banner. This interface is meant to provide a way
+ * to specify link actions and icon-based actions for a banner UI component.
+ */
+public interface BannerScope {
+    /**
+     * Configures a link action in the banner component.
+     *
+     * @param text The text label to display for the action.
+     * @param onClick The callback function to invoke when the action is clicked.
+     */
+    public fun action(text: String, onClick: () -> Unit)
+
+    /**
+     * Configures an icon-based action in the banner component.
+     *
+     * @param icon The icon to be displayed, represented as an `IconKey`.
+     * @param contentDescription A textual description for the icon, used for accessibility purposes.
+     *   Pass `null` if no description is needed.
+     * @param tooltipText Optional tooltip text to be displayed when the user hovers over the icon.
+     * @param onClick The callback function to be invoked when the icon is clicked.
+     */
+    public fun iconAction(icon: IconKey, contentDescription: String?, tooltipText: String? = null, onClick: () -> Unit)
+}
```

Comment:
I also thought about adding support for external links (similar to #3107) and actions that trigger a pop-up menu. Thoughts?